File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> storage;

        for (int i=0; i<s.length(); i++)
        {
            if (s[i]=='(' || s[i]=='{' || s[i]=='[')
                storage.push(s[i]);
            else
            {
                if (storage.empty())
                    return false;
                char tmp = storage.top();
                if (s[i]==')' && tmp!='(')
                    return false;
                else if (s[i]=='}' && tmp!='{')
                    return false;
                else if (s[i]==']' && tmp!='[')
                    return false;
                storage.pop();
            }
        }

        if (storage.empty())
            return true;
        return false;
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        close = {'}': '{', ']': '[', ')': '('}
        stack = []

        for c in s:
            if c not in close.keys():
                stack.append(c)
            else:
                if len(stack) == 0:
                    return False
                elif stack[-1] == close[c]:
                    stack.pop()
                else:
                    return False
        
        return len(stack) == 0